sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/UnsafeRowWriter.java

```diff
@@ -51,6 +51,7 @@ public UnsafeRowWriter(BufferHolder holder, int numFields) {
     this.nullBitsSize = UnsafeRow.calculateBitSetWidthInBytes(numFields);
     this.fixedSize = nullBitsSize + 8 * numFields;
     this.startingOffset = holder.cursor;
+    holder.reset();
```

Comment:
@cloud-fan @viirya For your worries, maybe we can move the `holder.reset()` to `BufferHolder`'s constructor. Then the holder will be reset only once, and also it'ok to continue writing from a buffer's current cursor.